variables.tf

```diff
@@ -332,3 +332,18 @@ variable "cluster_encryption_config" {
   }))
   default = []
 }
+
+variable "create_eks_fargate" {
+  description = "Controls if EKS Fargate resources should be created"
+  type        = bool
+  default     = false
+}
+
+variable "eks_fargate_profiles" {
+  description = "EKS Fargate profiles"
+  type = list(object({
```

Comment:
Use map of object please instead a list. More info here https://github.com/terraform-aws-modules/terraform-aws-eks/issues/476. And we are trying to move from count to for_each in all ressources.